app/screens/sign-in-screen/sign-in-screen.tsx

```diff
@@ -132,6 +135,7 @@ export class SignInScreen extends React.Component<SignInScreenProps, {}> {
     } catch (error) {
       logError(`Error occurs when signing in: ${error}`)
       Alert.alert(translate("signInScreen.error"), `${error}`)
+      this.props.userStore.authCore.setHasSignedIn(false)
```

Comment:
I think this should be handled inside `this._signInWithAuthCore()` or even `authCore.signIn()`?
The signIn flag is set inside those 2 functions